File: src/io/philbrick/minecraft/Packet.java
package io.philbrick.minecraft;

import java.io.*;

public class Packet extends ByteArrayInputStream {
    public int type;
    public int originalLen;
    public int compressedLen;

    public Packet(byte[] buf, int len) throws IOException {
        super(buf);
        originalLen = len;
    }

    public int readVarInt() throws IOException {
        return VarInt.read(this);
    }

    int readInteger() throws IOException {
        return Protocol.readInteger(this);
    }

    long readLong() throws IOException {
        return Protocol.readLong(this);
    }

    short readShort() throws IOException {
        return Protocol.readShort(this);
    }

    String readString() throws IOException {
        return Protocol.readString(this);
    }

    boolean readBoolean() {
        return read() != 0;
    }

    float readFloat() throws IOException {
        return Protocol.readFloat(this);
    }

    double readDouble() throws IOException {
        return Protocol.readDouble(this);
    }
}

File: src/main/scala/com/pygostylia/osprey/Packet.scala
package com.pygostylia.osprey

import java.io.{ByteArrayInputStream, InputStream}

class Packet(is: InputStream, val originalLen: Int) extends MinecraftInputStream(is) {
  val `type`: Int = readVarInt()

  def this(bs: Array[Byte]) = this(new ByteArrayInputStream(bs), bs.length)

  def this(bs: Array[Byte], originalLen: Int) = this(new ByteArrayInputStream(bs), originalLen)

  def packetType: Int = `type`
}